src/map/utils/guildutils.cpp

```diff
@@ -151,59 +143,70 @@ namespace guildutils
                 }
             }
         }
+
         ShowDebug("UpdateGuildsStock is finished");
     }
 
     void UpdateGuildPointsPattern()
     {
         // TODO: This function can be faulty when dealing with multiple processes. Needs to be synchronized properly across servers.
-        auto jstDayOfYear = earth_time::jst::get_yearday();
-        bool doUpdate     = static_cast<uint32>(serverutils::GetServerVar("[GUILD]pattern_update")) != jstDayOfYear;
+        const auto jstDayOfYear = earth_time::jst::get_yearday();
+        const bool doUpdate     = serverutils::GetServerVar("[GUILD]pattern_update") != jstDayOfYear;
 
         uint8 pattern = xirand::GetRandomNumber(8);
         if (doUpdate)
         {
             // write the new pattern and update time to try to prevent other servers from updating the pattern
             serverutils::SetServerVar("[GUILD]pattern_update", jstDayOfYear);
             serverutils::SetServerVar("[GUILD]pattern", pattern);
-            charutils::ClearCharVarFromAll("[GUILD]daily_points");
         }
         else
         {
             // load the pattern in case it was set by another server (and this server did not set it)
             pattern = serverutils::GetServerVar("[GUILD]pattern");
-            charutils::ClearCharVarFromAll("[GUILD]daily_points", true);
```

Comment:
PR looks good I'm just confused why this wasn't working. Deeper issue with `charutils::ClearCharVarFromAll` (or `charVarCache`)?